ui/page/governance/proposal_details_page.go

```diff
@@ -129,26 +131,34 @@ func (pg *ProposalDetails) HandleUserInteractions() {
 }
 
 func (pg *ProposalDetails) listenForSyncNotifications() {
+	if pg.ProposalNotificationListener == nil {
+		return
+	}
+	pg.ProposalNotificationListener = listeners.NewProposalNotificationListener()
+	err := pg.WL.MultiWallet.Politeia.AddNotificationListener(pg.ProposalNotificationListener, ProposalDetailsPageID)
+	if err != nil {
+		log.Errorf("Error adding politeia notification listener: %v", err)
+		return
+	}
 
 	go func() {
 		for {
-			var notification interface{}
-
 			select {
-			case notification = <-pg.Receiver.NotificationsUpdate:
-			case <-pg.ctx.Done():
-				return
-			}
-
-			switch n := notification.(type) {
-			case wallet.Proposal:
-				if n.ProposalStatus == wallet.Synced {
+			case notification := <-pg.ProposalNotifChan:
+				switch notification.ProposalStatus {
+				case wallet.Synced:
```

Comment:
Isn't an if statement here simpler and more straightforward?
```suggestion
				if notification.ProposalStatus == wallet.Synced {
```